modules/core/client/app/init.js

```diff
@@ -16,7 +16,7 @@ angular.module(ApplicationConfiguration.applicationModuleName).run(function ($ro
 
   // Check authentication before changing state
   $rootScope.$on('$stateChangeStart', function (event, toState, toParams, fromState, fromParams) {
-    if (toState.data && toState.data.roles && toState.data.roles.length > 0) {
+    if (toState.data && toState.data.roles && toState.data.roles.length > 0 && toState.data.roles.indexOf('guest') === -1) {
```

Comment:
Can this be moved to inside the forEach loop instead of the if statement? 

I'm not a fan of iterating over an array to check if we need to iterate over an array.